static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,100 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+
+import Jumbotron from '../components/Jumbotron';
+import PrivacyForm from '../components/PrivacyForm';
+import ProfileFormContainer from './ProfileFormContainer'
+import {
+  startProfileEdit,
+  updateProfile,
+  updateProfileValidation,
+  clearProfileEdit,
+  saveProfile,
+} from '../actions';
+
+class SettingsPage extends React.Component {
+
+  static propTypes = {
+    dispatch:       React.PropTypes.func.isRequired,
+    updateProfile:  React.PropTypes.func.isRequired,
+    saveProfile:    React.PropTypes.func.isRequired,
+    userProfile:    React.PropTypes.object.isRequired,
+    ui:             React.PropTypes.object.isRequired,
+    errors:         React.PropTypes.object,
+    isSettingPage:  React.PropTypes.bool
+  };
+
+  constructor(props) {
+    super(props);
+    this.updateProfile = this.updateProfile.bind(this);
+    this.saveProfile = this.saveProfile.bind(this);
+  }
+
+  render() {
+    const { userProfile: { profile }, ui, errors, isSettingPage } = this.props;
+    let preferredName = profile.preferredName || SETTINGS.name;
+
+    return (
+      <Jumbotron profile={ profile } text={ preferredName }>
+        <div className="card-copy">
+          <PrivacyForm {...this.props} />
+        </div>
+      </Jumbotron>
+    );
+  }
+
+  saveProfile(validator, profile, ui) {
+    const { dispatch } = this.props;
+    const username = SETTINGS.username;
+
+    // Validation errors will only show up if we start the edit
+    dispatch(startProfileEdit(username));
+
+    let errors = validator(profile, ui);
+    dispatch(updateProfileValidation(username, errors));
+    if (_.isEmpty(errors)) {
+      return dispatch(saveProfile(username, profile)).then(() => {
+        dispatch(clearProfileEdit(username));
+      });
+    } else {
+      return Promise.reject(errors);
+    }
+  }
+
+  updateProfile(profile) {
+    const { dispatch } = this.props;
+    const username = SETTINGS.username;
+    dispatch(startProfileEdit(username));
+    dispatch(updateProfile(username, profile));
+  }
+}
+
+const mapStateToProps = state => {
+  let errors;
+  let profile = {
+    profile: {}
+  };
+
+  if (state.profiles[SETTINGS.username] !== undefined) {
+    profile = state.profiles[SETTINGS.username];
+  }
+
+  if (profile.edit !== undefined) {
+    errors = profile.edit.errors;
+  } else {
+    errors = {};
+  }
+
+  console.log(state);
+  return {
+    userProfile: profile,
+    ui: state.ui,
+    isSettingPage: true,
+    errors: errors,
+    updateProfile: state.updateProfile,
+    saveProfile: state.saveProfile
```

Comment:
Are you sure this is a property of the state? I don't believe we can store functions in the redux store.